src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -732,6 +744,22 @@ private void ProcessPing(string targetNameOrAddress)
             return true;
         }
 
+        private IPHostEntry GetHostEntryWithCancel(string targetNameOrAddress)
+        {
+            var task = Dns.GetHostEntryAsync(targetNameOrAddress);
+            var waitHandles = new[] { ((IAsyncResult)task).AsyncWaitHandle, _dnsLookupCancel.Token.WaitHandle };
```

Comment:
This works fine, but how about the following?
```c#
try
{
    var task = Dns.GetHostEntryAsync(targetNameOrAddress);
    task.Wait(_dnsLookupCancel.Token);
    return task.
}
catch (OperationCanceledException e)
{
    throw new PipelineStoppedException();
}
```
Of course, this will require the change of `catch (Exception ex)` to deal with `AggregateException` specially by getting the first item from the `InnerExceptions` collection

Reply:
Poked at this one too, the aggregate exception is kinda annoying to have to deal with... Hard to really say one is better than the other myself, I don't know enough, but it seems like they're roughly equivalent?

Saw a few SO posts mentioning that the `GetAwaiter().GetResult()` pattern can result in nicer stacktraces as well, so we might want to stick to that, possibly? 🤔

Reply:
`GetAwaiter().GetResult()` unwraps the aggregate exception for you, basically give you the first item of the `InnerExceptions` collection.
In our case here, the aggregate exception will only have one item in its `InnerExceptions` collection. It has more than 1 inner exceptions only when the task is an aggregation of multiple tasks, for example, `Task.WhenAll`.

Reply:
This seems to indicate it's a bit more than just unwrapping: https://stackoverflow.com/questions/36426937/what-is-the-difference-between-wait-vs-getawaiter-getresult/36427080

But yes, there shouldn't be a huge difference here in either case, from what I'm reading, though I've seen a few heated discussions arguing about using one v.s. the other. 😁

Reply:
> This seems to indicate it's a bit more than just unwrapping:

Good point. I missed that :)